src/main/java/ch/jalu/configme/beanmapper/propertydescription/BeanDescriptionFactoryImpl.java

```diff
@@ -55,10 +65,12 @@ public class BeanDescriptionFactoryImpl implements BeanDescriptionFactory {
      * @param clazz the class to process
      * @return properties of the class
      */
-    protected @NotNull List<BeanPropertyDescription> collectAllProperties(@NotNull Class<?> clazz) {
-        List<PropertyDescriptor> descriptors = getWritableProperties(clazz);
+    public @NotNull List<BeanFieldPropertyDescription> getAllProperties(@NotNull Class<?> clazz) {
+        LinkedHashMap<String, Field> instanceFieldsByName = FieldUtils.getAllFields(clazz)
+            .filter(FieldUtils::isRegularInstanceField)
+            .collect(FieldUtils.collectByName(false));
```

Comment:
Final fields are not skipped here—might be wise to throw an exception if the field is final (without `@Ignore`)